user-story-nfdi4ing-instrument/functions.py

```diff
@@ -222,12 +233,35 @@ def generate_histogram_spec(data):
 
 
 def render_histogram(spec):
-    return(alt.Chart.from_dict(spec))
+    alt.Chart.from_dict(spec)
 
 
 def main(doi):
-    pass
+    # Instrument Metadata Display
+    metadata = get_metadata_display(doi)
+    view_metadata_display(metadata)
+
+    # Instrument connections list and histogram
+    data = get_events_by_doi_and_relation_type(doi, 'cites')
+    spec = generate_histogram_spec(data['meta']['occurred'])
+    # render_histogram(spec)
+
+    # Data that used an instrument
+    events = get_events_by_doi_and_relation_type(doi, 'cites') # Not sure what relation_type should be
+    formatted_citations = format_citations(events, doi)
+    view_html_table(formatted_citations)
+
+    # Publications that used an Instrument
+    events = get_events_by_doi_and_relation_type(doi, 'cites') # Not sure what relation_type should be
+    formatted_citations = format_citations(events, doi, include_authors=True)
+    view_html_table(formatted_citations)
+
+    # Co-authors List
+    view_authors_html_table(formatted_citations)
+
```

Comment:
What about having a HTML template that puts all the content in a nice order at the very end.